src/app/pages/quickMaps/pages/costEffectiveness/components/interventionCreation/interventionCreation.component.ts

```diff
@@ -1,37 +1,82 @@
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, AfterViewInit } from '@angular/core';
 import { DictionaryType } from 'src/app/apiAndObjects/api/dictionaryType.enum';
 import { InterventionsDictionaryItem } from 'src/app/apiAndObjects/objects/dictionaries/interventionDictionaryItem';
 import { Dictionary } from 'src/app/apiAndObjects/_lib_code/objects/dictionary';
 import { DialogData } from 'src/app/components/dialogs/baseDialogService.abstract';
 import { DialogService } from 'src/app/components/dialogs/dialog.service';
 import { QuickMapsService } from 'src/app/pages/quickMaps/quickMaps.service';
 import { DictionaryService } from 'src/app/services/dictionary.service';
+import { Location } from '@angular/common';
+import { Subscription } from 'rxjs';
+import { InterventionDataService } from 'src/app/services/interventionData.service';
 
 @Component({
   selector: 'app-intervention-creation',
   templateUrl: './interventionCreation.component.html',
   styleUrls: ['./interventionCreation.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
+  changeDetection: ChangeDetectionStrategy.Default,
 })
-export class InterventionCreationComponent {
+export class InterventionCreationComponent implements AfterViewInit {
   public interventionsDictionaryItems: Array<InterventionsDictionaryItem>;
   public selectedInterventions: Array<InterventionsDictionaryItem> = [];
+  public routerSubscription: Subscription;
 
   constructor(
     public quickMapsService: QuickMapsService,
     private dialogService: DialogService,
     private dictionariesService: DictionaryService,
+    private interventionService: InterventionDataService,
     private cdr: ChangeDetectorRef,
+    private location: Location,
   ) {
     void dictionariesService.getDictionaries([DictionaryType.INTERVENTIONS]).then((dicts: Array<Dictionary>) => {
       this.interventionsDictionaryItems = dicts.shift().getItems();
     });
   }
 
+  ngAfterViewInit() {
+    this.loadInterventions()
+  }
+
+  public async loadInterventions(): Promise<any> {
```

Comment:
return a Type, not `any`. If not returning anything return `void` e.g. `Promise<void>`